### PR TITLE
Add reference guide

### PR DESCRIPTION
**Pull request by svenvanderburg**
* Add API reference to user documentation
* Updates docstrings of client, publication, and namespaces modules.

### CODE REVIEW

File: examples/searching.ipynb [hunk 1]
```diff
@@ -2,7 +2,7 @@
  "cells": [
```
Comment (svenvanderburg) [inline examples/searching.ipynb:2]: I only added some cells for playing with `search_things`

Comment (raar1) [reply examples/searching.ipynb:2]: Yeah, ipynb files never play nice with git diff...

File: nanopub/client.py [hunk 2]
```diff
@@ -72,13 +89,20 @@ def find_nanopubs_with_pattern(self, subj=None, pred=None, obj=None,
...
+            type (str): A URI denoting the type of the introduced concept
+            searchterm (str): The term that you want to search on
+            max_num_results (int): Maximum number of result, default = 1000
+
         """
-        Searches the nanopub servers (at the specified grlc API) for any nanopubs of the given type, with given search term,
-        up to max_num_results.
-        """
-        if not type or not searchterm:
+        if not searchterm:
```
Comment (raar1) [inline nanopub/client.py:105]: Do we not need to check that type is also specified? If no longer true, then the error message should be clarified.

**Review by raar1 — APPROVED** [review]
Cool. Looks like docstrings are generally all up to date now.

File: nanopub/client.py (hunk 2 above)
Comment (svenvanderburg) [reply nanopub/client.py]: Good point. I removed the default for `type` argument which makes that check redundant. But indeed the error message is now misleading. I updated it. We should ideally do more validation, but I think this is fine for now.

**Pull request merged**